src/module.c

```diff
@@ -442,6 +444,29 @@ char *RM_Strdup(const char *str) {
     return zstrdup(str);
 }
 
+void RM_ProcessEventsWhileBlocked(RedisModuleCtx *ctx) {
+    long long now = getMonotonicUs();
+    long long elapsed = elapsedMs(ctx->start_time);
+
+    if (ctx->start_time == 0) {
+        ctx->start_time = ctx->next_event = now;
+        return;
+    }
```

Comment:
would be better to init this when the context is created rather than the first call.
let's merge it after #9890, it'll be easier.

i think we'll be able to do with just one variable (`next_event`), which will be set to `script_time_limit` when the context is created.